src/api/views.py

```diff
@@ -116,3 +118,45 @@ def oai(request):
     }
 
     return render(request, template, context, content_type="application/xml")
+
+
+def kbart(request):
+    """
+    Produces KBART metadata output
+    @param request: the request object
+    @return: a rendered CSV
+    """
+    fields = ["date_first_issue_online", "num_first_vol_online", "num_first_issue_online", "date_last_issue_online",
+              "num_last_vol_online", "num_last_issue_online", "title_url", "first_author", "title_id", "embargo_info",
+              "coverage_depth", "notes", "publisher_name", "publication_type", "date_monograph_published_print",
+              "date_monograph_published_online", "monograph_volume", "monograph_edition", "first_editor",
+              "parent_publication_title_id", "preceding_publication_title_id", "access_type"]
+
+    response = HttpResponse(content_type='text/csv')
+    writer = csv.DictWriter(response, fieldnames=fields)
+
+    writer.writeheader()
+
+    for journal in request.press.journals():
+        journal_line = {}
+        issues = journal.issues()
+        if issues.count() > 0:
+            journal_line['date_first_issue_online'] = '{:%Y-%m-%d}'.format(issues.order_by('date')[0].date)
+            journal_line['num_first_vol_online'] = issues.order_by('date')[0].volume
+            journal_line['num_first_issue_online'] = issues.order_by('date')[0].issue
+            journal_line['date_last_issue_online'] = '{:%Y-%m-%d}'.format(issues.order_by('-date')[0].date)
+            journal_line['num_last_vol_online'] = issues.order_by('-date')[0].volume
+            journal_line['num_last_issue_online'] = issues.order_by('-date')[0].issue
+
+        journal_line['title_url'] = journal.full_url()
+        journal_line['coverage_depth'] = 'fulltext'
```

Comment:
Will this be true of all journals?

Reply:
It may _technically_ be possible for a user to add a table of contents that points to non-free content using overlay functionality. Not sure how best we can cover that eventuality...

Reply:
Ignoring remote (`is_remote`) journals might be a good start.